cpp/src/sort/sort_column_impl.cuh

```diff
@@ -16,14 +16,68 @@
 
 #pragma once
 
-#include <sort/sort_impl.cuh>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/table/experimental/row_operators.cuh>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
 
 #include <thrust/sequence.h>
 #include <thrust/sort.h>
 
 namespace cudf {
 namespace detail {
 
+/**
+ * @brief Sort indices of a single column.
+ *
+ * This API offers fast sorting for primitive types. It cannot handle nested types and will not
+ * consider `NaN` as equivalent to other `NaN`.
+ *
+ * @tparam stable Whether to use stable sort
+ * @param input Column to sort. The column data is not modified.
+ * @param column_order Ascending or descending sort order
+ * @param null_precedence How null rows are to be ordered
+ * @param stable True if sort should be stable
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return Sorted indices for the input column.
+ */
+template <bool stable>
+std::unique_ptr<column> sorted_order(column_view const& input,
+                                     order column_order,
+                                     null_order null_precedence,
+                                     rmm::cuda_stream_view stream,
+                                     rmm::mr::device_memory_resource* mr);
```

Comment:
This is moved from `sort_impl.cuh`.

Reply:
I'm not sure that I see the benefit here. Before, `sort_column_impl.cuh` included `sort_impl.cuh`. Now, the situation is reversed. Why is one better than the other?

Reply:
`sort_impl.cuh` is sorting for table, which in turn calls sorting for column `sort_column_impl.cuh`. Thus, logically the order should be `sort_impl.cuh` include `sort_column_impl.cuh`.

Similarly, `(stable_)sort_column.cu` include `sort_column_impl.cuh`. We don't need them to include `sort_impl.cuh` in the dependency chain.